ui/store/constants/actions.ts

```diff
@@ -0,0 +1,101 @@
+export enum ActionConstants {
```

Comment:
@Gudahtt thoughts on this. IT makes sense as an enum for sure... just wanna get your take. 

@ryanml i'm not sure I am a fan of renaming this 'actions' as we already have an actions file. And the name of this enum is now a little weird. It is actually describing the possible values for the 'type' parameter of an action so ActionType might b e a better name.

Lets get @Gudahtt 's opinion here but I don't know that we need to make this an enum for this reason:

In my upcoming PR i migrate our store.js file to to ts and use the reduxjs/toolkit helper to build the store, which gives us the ability to derive the state and dispatch type from it. As we move more and more of our reducers to use reduxjs/toolkit we won't have a need for these action constants anymore and the dispatch method automatic typing will kick in for making sure we don't dispatch an action that doesn't exist in our reducers.

My opinion at this point is we should probably aim for a more minimal refactor to typescript, keeping the constants in the original casing, however I may be a bit biased here and I'm going to lean on Mark for a second opinion

Reply:
"action type" does seem like a better description of what this is. I don't have strong feelings either way on these other topics though.

Worth noting that even after we move everything to use Redux toolkit, we will have some cross-slice actions, so I expect we'll still have need of `actions.js` or something like it. But if we move to using `createAction` over manually created actions, that eliminates the need for these constants.

Reply:
@brad-decker I agree that ActionType would be a good name. I also don't have too strong of an opinion on the changes made in this PR.

I agree with the thought you shared in slack that "it doesn't take away anything from our ability to adopt toolkit"

Meanwhile, we could ask what are the benefits of the changes? What advantages does having these strings typed as enums give us over the present state on `develop`?

Reply:
If we were going to create an maintain our own type for the `type` key of actions it would have a ton of value, but as it stands I don't think we will. I'm going to sleep on it and think of the pros and cons a little more.